stats.py

```diff
@@ -0,0 +1,33 @@
+class WeatherStats:
+    def __init__(self):
+        self.highest = None
+        self.lowest = None
+        self.most_humid = None
+
+    def update_highest(self, temp, day):
+        if self.highest is None or temp > self.highest[0]:
+            self.highest = (temp, day)
+
+    def update_lowest(self, temp, day):
+        if self.lowest is None or temp < self.lowest[0]:
+            self.lowest = (temp, day)
+
+    def update_most_humid(self, humidity, day):
+        if self.most_humid is None or humidity > self.most_humid[0]:
+            self.most_humid = (humidity, day)
```

Comment:
Instead of using tuple to store day and temp, use a separate class. Make that class generic to handle day, temp, humidity etc.